src/shared/sleep-config.c

```diff
@@ -102,6 +102,88 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         return 0;
 }
 
+bool is_battery_low(void) {
+        _cleanup_free_ char *bat_cap_level = NULL;
+        int r;
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/capacity_level", &bat_cap_level);
+        if (r < 0) {
+                log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/capacity_level: %m");
+                return false;
```

Comment:
We usually return negative errno in such case.
Please change the return type from `bool` to `int`.

Reply:
sure. just a little detail here. You mean change the return type of the value returned here in case `r < 0`, right?

Reply:
there is only two possibility for this funtion : either yes battery is low or no battery is not low. It makes sense to keep bool in that way so i used bool type.
For cases to handle error, i think the debug_err message should be enough for detail... why change the complete return type?

Reply:
Could you please suggest some other way to handle that instead of changing the return type of entire function here?

Reply:
We usually (but not always, unfortunately(?)) make functions, that return boolean on success, return negative errno on failure, and leaves the error handling to the caller. E.g. `on_ac_power()` in udev-util.c returns negative errno on failure, and return 0 or 1 on success. 

I think it is better to do in the caller in general.
```
r = is_battery_low();
if (r < 0)
        log_warning_errno(r, "Failed to determine if battery is low, assuming not: %m");
if (r > 0) {
        OPERATIONS_WHEN_BATTERY_IS_LOW;
} else {
        OPERATIONS_WHEN_BATTERY_IS_CHARGED_OR_AC;
}
```

(BTW, I prefer `battery_is_low()` or `battery_capacity_rate_is_low()` instead of `is_battery_low()`, but this is not a strong opinion.)

Reply:
ok. Checked udev-util.c. It is a better practice. Changing the return type and function name.
Thanks.